app/models/name/create.rb

```diff
@@ -1,10 +1,15 @@
 # frozen_string_literal: true
 
 class Name < AbstractModel
+  scope :with_rank,
+        ->(rank) { where("`rank` = ?", Name.ranks[rank]) if rank }
+
```

Comment:
Would controllers/name_cntroller.rb be a better place for the `scope`?

Reply:
I makes more sense to me in the model than in the controller.  Having this scope in a controller would seem odd to me.